### PR TITLE
instruction_check: enhancement

### PR DESCRIPTION
**Pull request by haoliang-Zhu**
Depending on the platform, different test cases are executed now and some other optimizations to enhance the robustness of the code.

### CODE REVIEW

File: BM/instruction-check/feature_list.py [hunk 1]
```diff
@@ -1,51 +1,228 @@
...
+        "cpuid": ['7', '0', '1', '0', 'd', '4'],
+        "platforms": {"SRF", "CWF"}
+    },
+    "AVX_NE_CONVERT": {
+        "cpuid": ['7', '0', '1', '0', 'd', '5'],
+        "platforms": {"SRF", "CWF"}
+    },
+    "PREFETCHI": {
+        "cpuid": ['7', '0', '1', '0', 'd', '14'],
+        "platforms": {"SPR", "EMR", "GNR"}
```
Comment (xhao22) [inline BM/instruction-check/feature_list.py:170]: I don't think SPR/EMR support PREFETCHI.

File: BM/instruction-check/feature_list.py [hunk 2]
```diff
@@ -1,51 +1,228 @@
...
+        "cpuid": ['7', '0', '1', '0', 'a', '4'],
+        "platforms": {"SPR", "EMR", "GNR", "SRF", "CWF"}
+    },
+    "CMPCCXADD": {
+        "cpuid": ['7', '0', '1', '0', 'a', '7'],
+        "platforms": {"SRF", "CWF"}
+    },
+    "FRED": {
+        "cpuid": ['7', '0', '1', '0', 'a', '17'],
+        "platforms": {}
```
Comment (xhao22) [inline BM/instruction-check/feature_list.py:146]: What does the empty of platforms mean? Seems FRED only be supported on CWF and DMR.

File: BM/instruction-check/feature_list.py [hunk 3]
```diff
@@ -1,51 +1,228 @@
...
+    },
+    "WRMSRNS": {
+        "cpuid": ['7', '0', '1', '0', 'a', '19'],
+        "platforms": {"SRF", "CWF"}
+    },
+    "AMX_FP16": {
+        "cpuid": ['7', '0', '1', '0', 'a', '21'],
+        "platforms": {"SPR", "EMR", "GNR"}
```
Comment (xhao22) [inline BM/instruction-check/feature_list.py:154]: Seems SPR/EMR doesn't support AMX_FP16. 

Did you test your patch on these platform?

Comment (haoliang-Zhu) [reply BM/instruction-check/feature_list.py:154]: Thank you for your suggestion. I previously conducted tests on SPR, GNR, and SRF, but there may be some errors. I will double-check.

**Pull request merged**